### DISCUSSION TITLE
Umstieg von OpenDTU nach OpenDTU-on Battery

### DISCUSSION THREAD

**Discussion by flurfunk0868** [hoylabs/OpenDTU-OnBattery#542] [Q&A] [answered]
Hallo zusammen,
habe eigentlich nur ne kleine Frage: 
Da ich bereits eine funktionierende OpenDTU habe, würde ich gerne wissen ob ich alle Einstellungen wie Netzwerk/MQTT Konfigurationen verliere, wenn ich die OpenDTU-onbattery firmware drüber flashe?

Danke schonmal

**Comment by schlimmchen** [comment]
Mit Sicherheit kann ich das nicht sagen, allerdings würde ich nen Kasten Bier wetten, dass es geht. Dennoch: Mach doch einfach ein Backup deiner `config.json` (aus dem Web-Interface herunterladen über den Browser). Die kannst du im Zweifel unter OpenDTU-OnBattery wiederherstellen. Oder schlimmstenfalls wenn du zurück zur OpenDTU musst, weil etwas nicht klappt.

**Reply by flurfunk0868** [reply]
Ich gucke mal am Wochenende und werde berichten

**Reply by schlimmchen** [reply]
@flurfunk0868 Und? Hat's geklappt?

**Reply by ottelo9** [reply]
Würde mich auch interessieren. Habe das gleiche vor.

**Reply by Tiese** [reply, marked as answer]
Hallo,
ich hatte vor ein paar Wochen umgestellt.
Ging ohne Probleme - sogar die Daten waren **_alle_** noch da (der Tagesertrag natürlich nicht mehr)!

**Reply by mastertape** [reply]
Kann selbiges bestätigen...

**Reply by flurfunk0868** [reply]
Ich muss gestehen, ich hab einfach ne zweite DTU gebastelt und es gar nicht weiter probiert. Aber anscheinend funktioniert es ja bei einigen. Danke für die Antworten.

**Reply by Forstgrund** [reply]
Bei mir wurde alles übernommen Stand 25.01.2024...kämpfe allerdings mit dem Hichi Einbindung....
Kann mir einer helfen da?

**Reply by St31l** [reply]
Wie habt ihr denn gewechselt von OpenDTU zu OpenDTU onBattery? Geht das per Update over the air, als würde man ein normales Update von OpenDTU fahren?

**Reply by Forstgrund** [reply]
Ja genau....einfach über die Update Funktion

**Reply by St31l** [reply]
Alles klar, danke für die schnelle Rückmeldung.
Dann muss ich also noch schauen wie ich mein JK-BMS mit der DTU verbinde (da habe ich hier schon ein Anschlussdiagramm gesehen)

**Reply by schlimmchen** [reply]
@St31l https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Jikong-JK-BMS Melde dich gerne (neues Issue oder neue Diskussion), wenn etwas nicht klappt. Diese Anleitung sollte ausreichend sein. Einstellungen in der Web UI sollten hinreichend verständlich sein.

**Reply by St31l** [reply]
Moin schlimmchen,
alles klar mache ich.
Habe heute den Wechsel von OpenDTU zu onBattery durchgeführt und ich muss sagen der Umzug war klasse. Erst ein paar Fragezeichen, wie ich die Werte vom Shelly Pro3EM abgreife(MQTT oder HTTP und wie die Daten per IP richtig verarbeitet werden), aber da gibts in der Doku schon Beiträge, sodass schnell die richtigen Daten eingetragen werden konnten, das war klasse!
Wo ich mich noch etwas vor drücke ist die Änderung des PinOut zum Anschluss des JK BMS. Ich tendiere gerade zur simpelsten Lösung, also die OpenDTU Hardware direkt zum BMS zu versetzen und per UART-TTL ohne Umwege direkt anzubinden. Wenn ich das richtig verstanden habe wird dafür aber eine weitere Definition von Pins am ESP32 benötigt und ich sehe es schon komme, das mich das einige Stunden kosten wird bis ich das zum laufen bekomme, oder es wird danach garnicht mehr funktionieren (bin mit VS Code nicht sehr vertraut)

**Reply by schlimmchen** [reply]
Schön zu lesen! Also, wenn du ein JK BMS anschließen möchtest, kommst du *nicht* um das Hochladen eines [Device-Profiles](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles) drumherum. Zur Zeit arbeitest du also noch komplett ohne? Geht das denn überhaupt noch?

Du erstellst dein Device Profile, also deine pin_mapping.json, am PC, lädst die hoch, und wählst sie aus. Fertig. Das ist nicht kompliziert -- finde ich. Was ist denn deine Sorge, das dich Stunden kosten wird?

Mit UART direkt an die Pins des JK BMS mag simpel sein, aber in der Tat musst du dann recht nahe ran.

**Reply by joshia0** [reply]
Wie habt ihr denn gewechselt von OpenDTU zu OpenDTU onBattery? Geht das per Update over the air, als würde man ein normales Update von OpenDTU fahren?

Das würde mich auch interessieren.
Welche von den Dateien muß ich denn isntallieren 👍 
opendtu-onbattery-generic.bin
1.86 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic.factory.bin
1.92 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic_esp32.bin
1.86 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic_esp32.factory.bin
1.92 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic_esp32s3.bin
1.77 MB 2024-06-03T18:21:36Z
opendtu-onbattery-generic_esp32s3.factory.bin
1.84 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic_esp32s3_usb.bin
1.78 MB 2024-06-03T18:21:37Z
opendtu-onbattery-generic_esp32s3_usb.factory.bin
1.84 MB 2024-06-03T18:21:37Z
Source code (zip)
2024-06-03T18:15:56Z
Source code (tar.gz)

**Reply by Manos1966** [reply]
per Update over the air
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Upgrade-from-OpenDTU-to-OpenDTUonBattery

Haengt vom Board ab. Ich gehe aber davon aus, dass du ein ganz normales ESP32 Board hast.
Daher brauchst du die 
opendtu-onbattery-generic_esp32.bin
1.86 MB 2024-06-03T18:21:37Z
 Datei.

**Comment by mattberlinHA** [comment]
Hallo zusammen,

sorry, wenn ich so blöd frage, aber kann ich folgende openDTU einfach mit openDTU-onbattery flashen?

![Screenshot 2025-04-27 120232](https://github.com/user-attachments/assets/47909469-db80-4cec-be1c-57457a95fa47)

Und falls ja, dann das oberste Asset:
![Screenshot 2025-04-27 120412](https://github.com/user-attachments/assets/edc46b0e-8874-4efd-959b-22b215e878bd)

**Reply by Manos1966** [reply]
Ja, aber mit USB Kabel (nicht OTA).

Die Version die du brauchst:
.... 4MB no ota factory.bin

**Comment by mattberlinHA** [comment]
Danke für die schnelle Antwort. Puh, bei diesen Klimmzügen macht wohl das Investment in eine neuere DTU Sinn.

**Reply by Sorbat** [reply]
https://shop.allianceapps.io/products/allianceapps-opendtu-fusion
https://shop.semy3d.de/Fusion-23-16MB-DTU-Konfigurator

**Reply by mattberlinHA** [reply]
Ja, mir gefällt vor allem der LAN-Anschluss vom 2. Link!

Was ist eigentlich hiervon zu halten?:
https://www.dtushop.de/products/dtu-fur-hoymiles-hm-300-1500-lan-dtu-plug-play-mit-xxl-display-und-gehause-kopie?variant=53614494843225

**Reply by schlimmchen** [reply]
FYI: https://github.com/hoylabs/OpenDTU-OnBattery/?tab=readme-ov-file#%EF%B8%8F--about-hardware-for-sale

**Reply by Hannes58** [reply]
@mattberlinHA 
Ich stand vor der gleichen Frage und habe, da auch bei mir ein 4MB Board als OpenDTU im Einsatz ist, mich für die neuere Hardware wie in der Doku angegeben entschieden. Sobald die hier ankommt wird gebastelt,
Danke an alle, die hier so viel Herzblut investieren !

**Comment by wegomyway** [comment]
Hallöle zusammen, habe hier aufmerksam mitgelesen. Dachte ich mir "mach auch mal von openDTU zu openDTU on Battery flashen. irgendwie gescheitert.
Meine ist :
![Screenshot 2025-05-04 155635](https://github.com/user-attachments/assets/bcc3c585-81e6-4eff-a9db-3e43ab28ff67)
![Screenshot 2025-05-04 155716](https://github.com/user-attachments/assets/5dc652bc-3f97-4f2d-a1ee-3a319234a458)
Etwas weiter oben hab ich mich anhand der Hinweise auch soweit abarbeiten können. Über Web-Flasher auch korrekt gefunden, Firmware vom Book gewählt und kleines Hinweisfenster und los. Nach nedlosen 5 Minuten "gerödel" abgebrochen durch mich, ran ans Netzteil und eingeloggt und allesi st bei alten geblieben. Gibt es doch noch irgendwie Tipps/Tricks wie ich das erfolgreich bewerkstelligen kann ?
Schönen Sonntag im Mai aus Berlin

**Comment by Manos1966** [comment]
Hallo @wegomyway
Du hast einen ESP32 mit 4Mb, keinen ESP32-S3

> Very unfortunately, the Community WebFlasher currently has a maintenance issue. Consider it deprecated for ESP32 (non-S3), as it is unable to select the correct firmware binary blob for ESP32 chips.

You need to find another way to flash your ESP32
... and you have to use Firmware file [opendtu-onbattery-generic_esp32_4mb_no_ota.factory.bin](https://github.com/hoylabs/OpenDTU-OnBattery/releases/download/2025.04.17/opendtu-onbattery-generic_esp32_4mb_no_ota.factory.bin)

**Comment by Hannes58** [comment]
@schlimmchen 
Ich habe eine neue Hardware bekommen, die alte openDTU erst mal zur Seite gelegt und die neue in Betrieb genommen.
So weit so gut, aber ich bekomme keine Verbindung von keinem der HM-600 zur neuen openDTU hin. Offenbar will oder kann keiner der WR mit der neuen DTU reden. 
Update: Nach Anpassung der Polling-Time, die auf 5000 stand, und jetzt mal auf 5 verändert wurde, sehe ich endlich Live Data :-)
Aber: Die Leistung jedes WR ist minimal obwohl 100% eingestellt sind. Warum ist die Farbe jedes WR gelb ?
Ausserdem blinken derzeit die Leistungsindikatoren an den  WR rot ... 

![image](https://github.com/user-attachments/assets/fb9d8c50-add2-4783-9b8b-e705f7d72b37)


Nochmal edit:
Nach erneutem Restart aller Inverter ( also auch die DC eingänge mal kurz unterbrechen damit der inverter neu startet) klappt es jetzt.
![image](https://github.com/user-attachments/assets/262882d6-99d0-44e5-9d52-2b6da3eeb35c)

**Reply by schlimmchen** [reply]
> Update: Nach Anpassung der Polling-Time, die auf 5000 stand, und jetzt mal auf 5 verändert wurde, sehe ich endlich Live Data :-)

Kannst du ausschließen, dass du nach dem Wechsel zu OpenDTU-OnBattery nochmal zurück zu OpenDTU gewechselt bist, um dann wieder auf OpenDTU-OnBattery zu wechseln?

Dieser Fehler zeigt sich, wenn OpenDTU-OnBattery zweimal die Einheit des Polling-Intervals von Sekunden auf Millisekunden konvertiert.
